### PR TITLE
`Text exercises`: Remove feedback conflict badge and views

### PR DESCRIPTION
**Pull request by pal03377**
#### General

- [X] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

The feedback conflict/consistency feature causes high coupling between Artemis and Athena. The feature hasn't been in use in production, apparently: There are no feedback conflict entries present at all on the production servers.

Removing this feature will make the integration of Athena-New and dropping the old system much easier. We will opt for a decoupled system

### Description

Changes affecting the text exercise assessment mode:
- Remove `Feedback Conflicts` warning badge in the textblock feedback editor
- Remove Feedback Conflict Resolve page

Endpoints:
- Removed endpoint `/participations/{participationId}/submissions/{submissionId}/feedbacks/{feedbackId}/feedback-conflicts`
- Removed endpoint `/exercises/{exerciseId}/feedback-conflicts/{feedbackConflictId}/solve`

Client routes:
- Removed page on `/course-management/{courseId}/text-exercises/{exerciseId}/participations/{participationId}/submissions/{submissionId}/text-feedback-conflict/{feedbackId}`
- Removed page on `/course-management/{courseId}/exams/{examId}/text-exercises/{exerciseId}/participations/{participationId}/submissions/{submissionId}/text-feedback-conflict/{feedbackId}` (corresponding exam page)

Database:
- The database table related to this feature will be removed in #6774.
- This way, this PR (#6769) can be tested on test servers without modifying their database.

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Student
- 1 Text exercise

1. As a student, create a submission on the text exercise (also possible as the tutor)
2. As a tutor, go to the Exercise Dashboard, and click "Start participating in the exercise" (in blue). Then start assessing your own submission from step 1
3. Give some feedback on the exercise
4. Submit the assessment
5. Go back to the exercise dashboard
6. Open the assessment again.
7. Ensure that there are no new errors in the developer console
8. Ensure that nothing breaks in the process
9. Try some maybe unexpected stuff around assessing the text exercise

#### Performance Review
- [X] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [X] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
Please pay some additional attention to whether I really only removed the "conflict" related parts of code related to **text exercises**. Many more code snippets are using the name "conflict" in the Artemis code base (e.g., programming exercise Git repository conflicts).
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [feedback-conflict.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/entities/feedback-conflict.ts.html) | deleted |  |
| [feedback.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/entities/feedback.model.ts.html) | 95.5% | ✅ |
| [text-assesment-event.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/entities/text-assesment-event.model.ts.html) | 100% | ✅ |
| [text-feedback-conflicts-header.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/conflicts/conflicts-header/text-feedback-conflicts-header.component.ts.html) | deleted |  |
| [text-feedback-conflicts.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/conflicts/text-feedback-conflicts.component.ts.html) | deleted |  |
| [text-assessment-area.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-assessment-area/text-assessment-area.component.ts.html) | 96.15% | ✅ |
| [text-assessment-base.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-assessment-base.component.ts.html) | 71.42% | ✅❌ |
| [text-assessment.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-assessment.service.ts.html) | 86.36% | uncovered code will be removed in #6764 anyways |
| [text-submission-assessment.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-submission-assessment.component.ts.html) | 90.42% | ✅ |
| [text-submission-assessment.route.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/text-submission-assessment.route.ts.html) | 80% | ✅❌ |
| [textblock-assessment-card.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/textblock-assessment-card/textblock-assessment-card.component.ts.html) | 82.5% | ✅❌ |
| [textblock-feedback-editor.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/assess/textblock-feedback-editor/textblock-feedback-editor.component.ts.html) | 94.8% | ✅ |
| [navbar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-6/artifact/shared/Coverage-Report-Client-Tests/app/shared/layouts/navbar/navbar.component.ts.html) | 89.89% | ✅❌ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [FeedbackConflict.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/FeedbackConflict.html) | deleted |  |
| [EscalationState.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.enumeration/EscalationState.html) | deleted |  |
| [FeedbackConflictType.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.enumeration/FeedbackConflictType.html) | deleted |  |
| [TextAssessmentEventType.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.enumeration/TextAssessmentEventType.html) | 100% | ✅ |
| [FeedbackConflictRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/FeedbackConflictRepository.html) | deleted |  |
| [AutomaticTextAssessmentConflictService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/AutomaticTextAssessmentConflictService.html) | deleted |  |
| [ResultService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ResultService.html) | 91% | ✅ |
| [TextAssessmentService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/TextAssessmentService.html) | 100% | ✅ |
| [TextAssessmentConflictService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.athene/TextAssessmentConflictService.html) | deleted |  |
| [FeedbackConflictResponseDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dto/FeedbackConflictResponseDTO.html) | deleted |  |
| [TextFeedbackConflictRequestDTO.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dto/TextFeedbackConflictRequestDTO.html) | deleted |  |
| [TextAssessmentResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5050-4/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/TextAssessmentResource.html) | 90% | ✅ |

### Screenshots

**Before**: Feedback conflict warning badge (screenshot from a thesis, hard to reproduce without Athena running)
screenshot-2023-06-26_001465
**After:** The badges never appear:
screenshot-2023-06-26_001467

**Before:** Feedback conflict resolve view (screenshot from a thesis, hard to reproduce without Athena running)
screenshot-2023-06-26_001466
**After:** (removed)

### CODE REVIEW

**Review by FelixTJDietrich — DISMISSED** [review]
Great job on cleanly removing everything related to the feature. I looked around and could not find any other remains :) The added tests also seem okay to me, I don't know how sensible it is to add more in this PR. Code LGTM, also tested on TS4 during a testing session and everything works as before. Approved!

**Review by TheZoker — APPROVED** [review]
Tested during testing session on TS4 and LGTM 👍

**Review by simon-hng — APPROVED** [review]
Tested during testing session

**Review by greekgeek1 — DISMISSED** [review]
Tested on ts3, works as expected

**Review by FelixTJDietrich — APPROVED** [review]
Code looks good to me after resolving the merge conflicts 🎉

File: src/main/webapp/app/exercises/text/assess/text-assessment.service.ts [hunk 1]
```diff
@@ -9,7 +9,6 @@ import { StudentParticipation } from 'app/entities/participation/student-partici
 import { TextBlock } from 'app/entities/text-block.model';
 import { TextBlockRef } from 'app/entities/text-block-ref.model';
 import { TextSubmission } from 'app/entities/text-submission.model';
```
Comment (terlan98) [inline src/main/webapp/app/exercises/text/assess/text-assessment.service.ts:11]: Unused import

```suggestion
```

**Review by terlan98 — CHANGES_REQUESTED** [review]
The changes look good to me 👍  I also could not find any remaining feedback-conflict-related parts. 

There's just 1 unused import. It'd be nice if you could remove that.

**Review by terlan98 — APPROVED** [review]
Thanks for removing the unused import 🙂  Approved

**Review by FelixTJDietrich — APPROVED** [review]

**Review by simon-hng — APPROVED** [review]

**Review by TheZoker — APPROVED** [review]

**Review by maximiliansoelch — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**